torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -656,25 +661,87 @@ FusionExecutor::GlobalBuffers FusionExecutor::allocGlobalVals(
 }
 
 std::vector<at::Tensor> FusionExecutor::allocOutputs(
-    const at::ArrayRef<IValue>& inputs,
     kir::ExpressionEvaluator& expr_eval,
     const std::unordered_set<int>& alias_indices) {
   FUSER_PERF_SCOPE("FusionExecutor::AllocOutputs");
   const auto kernel = lowered_->kernel();
   // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
   std::vector<at::Tensor> outputs;
+  for (const auto out_i : c10::irange(kernel->outputs().size())) {
+    // TODO: FIX this short-cut
+    // If the output is just trivially the input, just "copy" it over.
+    if (kernel->outputs()[out_i]->isFusionInput()) {
+      TORCH_INTERNAL_ASSERT(false, "trivial input forwarding NOT IMPLEMENTED");
```

Comment:
There is some subtle difference. The trivial forward here needs to be kept as real output, versus aliased output is only used for updating on input tensor and should be dropped.

Currently I don't see the trivial forward hooked up when we allocate outputs, so I suspect it just isn't working at all. And it's also not tested anywhere :laughing: 

it can be easily made to work though. I just want to double check that codegen API actually does support marking input tensor as output. I'll handle it in a follow up PR.

Reply:
> aliased output is only used for updating on input tensor and should be dropped

Does the kernel not need it to be plumbed through as an output? I thought the output name would be different in the kernel than the input name so we'd still need to make sure it's positioned correctly in the kernel arguments.

Reply:
Yes that's how it worked within the kernel. We do have output with different names that points to the same memory.

But when we collect outputs to be returned to framework, we strip aliased outputs away. i.e. BN doesn't return running stats as outputs but just silently update them.
Unlike here in the forwarded output, where they do get returned to framework as outputs from fusion.